app/push/server/push.js

```diff
@@ -128,6 +128,11 @@ export class PushClass {
 				return;
 			}
 
+			if (response?.statusCode === 401) {
```

Comment:
I guess this is one of the bad habits that optional chaining introduces, with this change the code is now checking twice if `response` is defined.. IMO we should avoid this and in this case we could use a single `if` to check if `response.statusCode` exists:

```javascript
			if (response?.statusCode) {
				if (response.statusCode === 406) {
					logger.info('removing push token', token);
					appTokensCollection.remove({
						$or: [{
							'token.apn': token,
						}, {
							'token.gcm': token,
						}],
					});
					return;
				}

				if (response.statusCode === 401) {
					logger.warn('Error sending push to gateway (not authorized)', response);
					return;
				}
			}
...
```